Components/Sensors/BME280.cs

```diff
@@ -0,0 +1,367 @@
+        {
+            this.IsSPI = true;
+            this.SPIBus = SPIBus;
+            this.SPICS = ChipSelect;
+        }
+
+        /// <summary> Sets the device's registers to match the given <c>Configuration</c>. </summary>
+        /// <param name="Configuration"> The configuration to apply. </param>
+        public void Configure(Config Configuration)
+        {
+            this.Configuration = Configuration;
+            this.CompParams = ReadCompVals();
+            ChangeMode(Mode.SLEEP);
+
+            byte[] RawData = Read((byte)Register.CTRL_HUM, 4);
+
+            byte osrs_h = (byte)(Configuration.MeasureHumidity ? (byte)Configuration.HumidityOversampling : 0);
+            byte Ctrl_Hum = (byte)((RawData[0] & 0b1111_1000) | osrs_h & 0b111);
+
+            byte osrs_t = (byte)Configuration.TemperatureOversampling;
+            byte osrs_p = (byte)(Configuration.MeasurePressure ? (byte)Configuration.PressureOversampling : 0);
+            byte Ctrl_Meas = (byte)(((osrs_t << 5) & 0b1110_0000) | ((osrs_p << 2) & 0b0001_1100) | ((byte)Mode.SLEEP & 0b0000_0011)); // Stays in SLEEP mode, we'll exit once these writes are all done.
+
+            byte Config = (byte)((((byte)Configuration.StandbyTime << 5) & 0b1110_0000) | (((byte)Configuration.IIRFilterTimeConstant << 2) & 0b0001_1100) | (RawData[3] & 0b0000_0010) | (Configuration.Use3WireSPI ? 1 : 0));
+
+            Write((byte)Register.CTRL_HUM, Ctrl_Hum);
+            Write((byte)Register.CTRL_MEAS, Ctrl_Meas);
+            Write((byte)Register.CONFIG, Config);
+
+            ChangeMode(Configuration.Mode);
+        }
+
+        /// <summary> Configures the device with default settings, which should be good enough for basic functionality. </summary>
+        public void Configure() => Configure(DefaultConfig);
+
+        /// <summary> Changes the acquisition mode of the device, or brings it in/out of SLEEP. </summary>
+        /// <param name="NewMode"> The <c>Mode</c> to put the device into. </param>
+        public void ChangeMode(Mode NewMode)
+        {
+            byte Config = Read((byte)Register.CTRL_MEAS, 1)[0];
+            Config = (byte)((Config & 0b1111_1100) | ((byte)NewMode & 0b0000_0011));
+            Write((byte)Register.CTRL_MEAS, Config);
+        }
+
+        /// <summary> Gets the sensor's current readings in an easy to store format. </summary>
+        public DataUnit GetData()
+        {
+            return new DataUnit("BME280")
+            {
+                { "Temperature", this.Temperature },
+                { "Pressure", this.Pressure },
+                { "Humidity", this.Humidity }
+            }.SetSystem(this.System);
+        }
+
+        /// <summary> Checks if the device is responding by querying a known, constant register value. </summary>
+        public bool Test()
+        {
+            byte[] DeviceData = Read((byte)Register.DEV_ID, 1);
+            return (DeviceData != null) && (DeviceData.Length > 0) && (DeviceData[0] == 0x60);
```

Comment:
Theoretically, should this be `DeviceData.Length == 1`?

Reply:
I was leaving the possibility of it returning more data open, just in case. We only care about the first byte being 0x60, but anything else doesn't matter.